app/controllers/application_controller.rb

```diff
@@ -3,6 +3,10 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
+  before_filter do |c|
+    User.current_user_id = User.find(c.session[:user]).id unless c.session[:user].nil?
+  end
```

Comment:
@nowakowski Piotr, I believe the model.table_data method is called from a single place only, that is a single line inside the DataTablesController. That means you might be able to simply pass current_user as a parameter there, relieving User class from a surplus class variable and not making the above query, unnecessary, for any other non-data-table call. You will then be able to pass that current_user value down to 'visible' scopes, likewise. Wouldn't that be a better solution?